src/main.py

```diff
@@ -0,0 +1,704 @@
+
+
+def main_game_loop(**kwargs):  # pylint: disable=unused-argument
+    game_running = True
+    player_turn_index = 0
+    current_player = players[player_turn_index]
+    dice_rolled.append((current_player.roll_dice(2),
+                        current_player.roll_dice(2)))
+
+    pprint(board)
+    # GAME LOOP
+    # 1 LOOP THROUGH PLAYERS AND INITIALISE STARTING POSITIONS.
+    # 2 BEGIN MAIN TURN BASED LOOP.
+    # 2.1 PLAYER HAS CHOICE OF BUILDING AND
+
+    while game_running:
+        for event in pygame.event.get():  # pylint: disable=no-member
+            if event.type == pygame.QUIT:  # pylint: disable=no-member
+                game_running = False
+
+            elif event.type == pygame.KEYDOWN:  # pylint: disable=no-member
+                if event.key == pygame.key.K_SPACE:
+                    if player_turn_index == len(players) - 1:
+                        player_turn_index = 0
+                        current_player = players[player_turn_index]
+                    else:
+                        player_turn_index += 1
+                        current_player = players[player_turn_index]
+
+                    dice_roll1, dice_roll2 = current_player.roll_dice(2)
+                    dice_rolled.append((dice_roll1, dice_roll2))
+
+                    for game_tile in board:
+                        if dice_roll1 + dice_roll2 == game_tile.real_number:
+                            current_player.add_resources(game_tile)
+
+                            card = game_tile.tile.generate_resource().name()
+                            game_log_txt = ''.join(
+                                f'{current_player.name} just rolled a '
+                                f'{dice_roll1+dice_roll2}. Added '
+                                f'{card} to inventory'
+                            )
+
+                            game_log.append(game_log_txt)
+
+            elif event.type == pygame.MOUSEBUTTONDOWN:  # pylint: disable=no-member # nopep8 E501
+                click_event(event, current_player)  # handles clicking events
+
+            elif event.type == pygame.MOUSEMOTION:  # pylint: disable=no-member
+                mouse_motion_event()  # function handles mouse motion events
+
+        if check_player_won():
+            # draw_end_screen() # TODO
+            game_running = False
+
+        draw(current_player)
+        for line in built_roads:
+            pygame.draw.line(
+                screen,
+                line[2].color,
+                (line[0].x_pos, line[0].y_pos),
+                (line[1].x_pos, line[1].y_pos),
+                5)
+        draw_buildings()  # draw_settlements
+        draw_buildings(city=True)  # draw cities
+
+        # count how many settlements
+        # each player has and award vp accordingly
+
+        # player_longest_road = calc_longest_road()
+        # player_longest_road.victory_points +=1
+        DrawDice.draw(screen, dice_rolled[-1][0], dice_rolled[-1][1])
```

Comment:
theres a bug here i think why is dice_rolled a list of a single tuple, with another 2 tuples inside with 2 values each, i thought the whole point of dice_rolled was to be a list of 2 values from the two dice rolls

Reply:
yeah I had the same issue yesterday when trying to run it, was gonna change it to be just the 2 values but not sure if @EddieJ01 had something else in mind